util/report_utils/build_scale_csv_summary.rb

```diff
@@ -0,0 +1,169 @@
+# frozen_string_literal: true
+
+require "../../tests/helpers/perf_results_helper.rb"
+include PerfResultsHelper # rubocop:disable Style/MixinUsage
+
+raise Exception, "you must provide a results directory" unless ARGV[0]
+
+parent_dir = ARGV[0]
+
+TEMPLATE_PATH = "./templates/scale_csv_summary_template.html"
+
+SUMMARY_CSV_HEADINGS = ["result",
+                        "agents",
+                        "ok",
+                        "ko",
+                        "combined mean",
+                        "catalog mean",
+                        "filemeta plugins mean",
+                        "filemeta pluginfacts mean",
+                        "locales mean",
+                        "node mean",
+                        "report mean",
+                        "average CPU %",
+                        "average memory"].freeze
+
+@template_path = ENV["TEMPLATE_PATH"] || TEMPLATE_PATH
+
+def process_results_csv(csv_path, summary_csv_path)
+  # ensure the expected result csv file exists
+  raise "File not found: #{csv_path}" unless File.file?(csv_path)
+
+  # create a csv file with the headings and last 2 rows
+  output_path = csv_path.gsub(".csv", ".scale_extract.csv")
+  csv_name = File.basename(csv_path)
+  result_name = csv_name.gsub(".csv", "")
+  puts "processing csv: #{csv_name}"
+
+  # TODO: extract into a separate method?
+  contents = File.readlines(csv_path)
+  File.open(output_path, "w") do |f|
+    f << contents[0]
+    f << contents[contents.length - 2]
```

Comment:
there is no need to use `contents.length` here.  Ruby arrays can be addressed by negative indexes. These lines can be:
```
f << contents[- 2]
f << contents[- 1]
```

Reply:
Very cool! Updated...